src/rust/multi.bato/src/parser.rs

```diff
@@ -0,0 +1,339 @@
+				status: MangaStatus::Unknown,
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Default,
+			});
+		}
+	}
+}
+
+pub fn parse_manga(obj: Node, id: String) -> Result<Manga> {
+	let title = obj.select(".item-title").text().read();
+	let cover = obj.select(".shadow-6").attr("src").read();
+	let description = obj.select(".limit-html").text().read();
+
+	let mut author = String::new();
+	let mut artist = String::new();
+	let mut status_str = String::new();
+	let mut categories: Vec<String> = Vec::new();
+
+	for i in obj.select(".attr-item").array() {
+		let item = i.as_node();
+		if item.select("b").text().read().contains("Author") {
+			author = item.select("span").text().read();
+		}
+		if item.select("b").text().read().contains("Artist") {
+			artist = item.select("span").text().read();
+		}
+		if item.select("b").text().read().contains("Original") {
+			status_str = item.select("span").text().read();
+		}
+		if item.select("b").text().read().contains("Genres") {
+			let gen_str = item.select("span").text().read();
+			let split = gen_str.as_str().split(',');
+			let vec = split.collect::<Vec<&str>>();
+			for item in vec {
+				categories.push(String::from(item.trim()));
+			}
+		}
+	}
+
+	let mut url = String::new();
+	if let Ok(url_str) = defaults_get("sourceURL").as_string() {
+		url.push_str(url_str.read().as_str());
+		url.push_str("/series/");
+		url.push_str(&id);
+	}
+
+	let status = if status_str.contains("Ongoing") {
+		MangaStatus::Ongoing
+	} else if status_str.contains("Completed") {
+		MangaStatus::Completed
+	} else if status_str.contains("Hiatus") {
+		MangaStatus::Hiatus
+	} else if status_str.contains("Cancelled") {
+		MangaStatus::Cancelled
+	} else {
+		MangaStatus::Unknown
+	};
+
+	let mut nsfw = MangaContentRating::Safe;
+	if !obj
+		.select(".alert.alert-warning span b")
+		.text()
+		.read()
+		.is_empty()
+	{
+		nsfw = MangaContentRating::Nsfw;
+	}
+
+	Ok(Manga {
+		id,
+		cover,
+		title,
+		author,
+		artist,
+		description,
+		url,
+		categories,
+		status,
+		nsfw,
+		viewer: MangaViewer::Scroll,
+	})
+}
+
+pub fn get_chaper_list(obj: Node) -> Result<Vec<Chapter>> {
+	let mut chapters: Vec<Chapter> = Vec::new();
+	for item in obj.select(".item").array() {
+		let chapter_node = item.as_node();
+		let id = chapter_node
+			.select("a")
+			.attr("href")
+			.read()
+			.replace("/chapter/", "");
+		let title = chapter_node.select(".chapt span").text().read();
+		let name = chapter_node
+			.select("a b")
+			.text()
+			.read()
+			.replace("Chapter", "");
+		let _time_str = chapter_node.select(".extra i.ps-3").text().read();
+
+		let chapter = String::from(name.trim()).parse::<f32>().unwrap_or(-1.0);
+		if let Ok(url_str) = defaults_get("sourceURL").as_string() {
+			let mut url = url_str.read();
+			url.push_str("/chapter/");
+			url.push_str(&id);
+
+			chapters.push(Chapter {
+				id,
+				title,
+				volume: -1.0,
+				chapter,
+				date_updated: -1.0,
+				scanlator: String::new(),
```

Comment:
you can get both uploader name and the date updated (though the date is relative so it takes a little more effort to parse)

Reply:
Added parsing for date and scanlator label in two commits. [875f7d8](https://github.com/Skittyblock/aidoku-community-sources/pull/114/commits/875f7d8e27dd5143bf28c64619cff651a0f68363) and [f63a5e1](https://github.com/Skittyblock/aidoku-community-sources/pull/114/commits/f63a5e151617e96b1e8c8228f67952afd811e72a)